### PR TITLE
Makefile: add support for TARGETPLATFORM variable

### PR DESCRIPTION
**Pull request by barkbay**
Unless I'm missing something it is not possible to create `linux/amd64` Docker images on my `arm64` laptop because `TARGETPLATFORM` is never set: https://github.com/elastic/cloud-on-k8s/blob/3901aca32b4e47124ed3dc7546fedda737efb10b/Dockerfile#L2

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -396,12 +396,15 @@ else
...
+TARGETPLATFORM ?= linux/amd64
+
 docker-build: go-generate generate-config-file 
 	DOCKER_BUILDKIT=1 docker build . \
 		--progress=plain \
 		--build-arg GO_LDFLAGS='$(GO_LDFLAGS)' \
 		--build-arg GO_TAGS='$(GO_TAGS)' \
 		--build-arg VERSION='$(VERSION)' \
+		--build-arg TARGETPLATFORM='$(TARGETPLATFORM)' \
 		-t $(OPERATOR_IMAGE)
```
[the remark this answers is not in the document]
Comment (barkbay) [reply Makefile:408]: > `TARGETPLATFORM` should be automatically set when the Docker

Yes, it is the case. But when I generate a Docker image it's mostly to deploy it on `amd64` nodes (for example GKE), hence building `arm64` is useless most of the time.

```diff
+		--platform $(BUILD_PLATFORM) \
```

This fails with:

```
DOCKER_BUILDKIT=1 docker build . \
                --progress=plain \
                --build-arg GO_LDFLAGS='-X github.com/elastic/cloud-on-k8s/v2/pkg/about.version=2.5.0-SNAPSHOT -X github.com/elastic/cloud-on-k8s/v2/pkg/about.buildHash=59698458 -X github.com/elastic/cloud-on-k8s/v2/pkg/about.buildDate=2022-08-30T08:42:22Z -X github.com/elastic/cloud-on-k8s/v2/pkg/about.buildSnapshot=true' \
                --build-arg GO_TAGS='' \
                --build-arg VERSION='2.5.0-SNAPSHOT' \
                --platform "linux/amd64,linux/arm64" \
                -t gcr.io/elastic-cloud-dev/eck-operator-michael-dev/eck-operator-michael:latest
Error response from daemon: "amd64,linux" is an invalid component of "linux/amd64,linux/arm64": platform specifier component must match "^[A-Za-z0-9_-]+$": invalid argument
make: *** [docker-build] Error 1
```

[the remark this answers is not in the document]
Comment (thbkrkr) [reply Makefile:408]: Oh, surprisingly `DOCKER_BUILDKIT=1 docker build` only supports one platform compared to `docker buildx`, not a list. and this is independently of the fact of using `--platform xxx` or like you propose `--build-arg TARGETPLATFORM=xxx`.

`TARGETPLATFORM=linux/arm64,linux/amd64 make docker-build` fails the same.

With my proposal you can do `BUILD_PLATFORM=linux/amd64 make docker-build`. We avoid to introduce a new env var.

[the remark this answers is not in the document]
Comment (thbkrkr) [reply Makefile:408]: > If BUILD_PLATFORM is not set (by the user/developer) I think I would then expect the command to succeed ?

Last try. We could change the default?
`docker-multiarch-build` is only used by `ci-release`, called after `setenvconfig dev/build|build`.

```diff
diff --git a/.ci/setenvconfig b/.ci/setenvconfig
index d334d4729..03836a2ba 100755
--- a/.ci/setenvconfig
+++ b/.ci/setenvconfig
@@ -537,7 +537,6 @@ E2E_JSON=true
 E2E_PROVIDER=kind
 CLUSTER_NAME=eck-pr-$BUILD_NUMBER
 TEST_OPTS=-race
-BUILD_PLATFORM=linux/amd64
 
 ENV
 write_deployer_config <<CFG
@@ -582,6 +581,7 @@ export LICENSE_PUBKEY=/go/src/github.com/elastic/cloud-on-k8s/.ci/license.key
 
 SNAPSHOT=true
 IMG_VERSION=$(cat $WORKSPACE/VERSION)-$(date +%F)-$(git rev-parse --short --verify HEAD)
+TARGET_PLATFORM=linux/amd64,linux/arm64
 ENV
 
   else
@@ -599,6 +599,7 @@ export LICENSE_PUBKEY=/go/src/github.com/elastic/cloud-on-k8s/.ci/license.key
 
 SNAPSHOT=false
 IMG_VERSION=$TAG_NAME
+TARGET_PLATFORM=linux/amd64,linux/arm64
 ENV
 
   fi
@@ -615,6 +616,7 @@ E2E_REGISTRY_NAMESPACE=eck-ci
 IS_SNAPSHOT_BUILD=yes
 GO_TAGS=release
 export LICENSE_PUBKEY=/go/src/github.com/elastic/cloud-on-k8s/.ci/license.key
+TARGET_PLATFORM=linux/amd64,linux/arm64
 ENV
 ;;
 
diff --git a/Makefile b/Makefile
index 2cc03196c..303369482 100644
--- a/Makefile
+++ b/Makefile
@@ -356,7 +356,7 @@ switch-tanzu:
 ##  --    Docker images    --  ##
 #################################
 
-BUILD_PLATFORM ?= "linux/amd64,linux/arm64"
+TARGET_PLATFORM ?= linux/amd64
 
 docker-multiarch-build: go-generate generate-config-file 
 ifeq ($(SNAPSHOT),false)
@@ -367,7 +367,7 @@ ifeq ($(SNAPSHOT),false)
 		--build-arg GO_LDFLAGS='$(GO_LDFLAGS)' \
 		--build-arg GO_TAGS='$(GO_TAGS)' \
 		--build-arg VERSION='$(VERSION)' \
-		--platform $(BUILD_PLATFORM) \
+		--platform $(TARGET_PLATFORM) \
 		-t $(OPERATOR_IMAGE) \
 		-t $(OPERATOR_DOCKERHUB_IMAGE) \
 		--push
@@ -390,7 +390,7 @@ else
 		--build-arg GO_LDFLAGS='$(GO_LDFLAGS)' \
 		--build-arg GO_TAGS='$(GO_TAGS)' \
 		--build-arg VERSION='$(VERSION)' \
-		--platform $(BUILD_PLATFORM) \
+		--platform $(TARGET_PLATFORM) \
 		-t $(OPERATOR_IMAGE) \
 		--push
 endif
@@ -402,6 +402,7 @@ docker-build: go-generate generate-config-file
 		--build-arg GO_LDFLAGS='$(GO_LDFLAGS)' \
 		--build-arg GO_TAGS='$(GO_TAGS)' \
 		--build-arg VERSION='$(VERSION)' \
+		--platform $(TARGET_PLATFORM) \
 		-t $(OPERATOR_IMAGE)
 
 docker-push:
@@ -464,7 +465,7 @@ e2e-docker-multiarch-build: go-generate
 		--file test/e2e/Dockerfile \
 		--build-arg E2E_JSON=$(E2E_JSON) \
 		--build-arg E2E_TAGS='$(E2E_TAGS)' \
-		--platform $(BUILD_PLATFORM) \
+		--platform $(TARGET_PLATFORM) \
 		--push \
 		-t $(E2E_IMG) .
```

**Comment by barkbay** [comment]
Closing as I'm not sure I'll have the time to work on it soon, I'll reopen later if needed. Sorry for the noise.

**Pull request closed without merging**